popde/density_estimate.py

```diff
@@ -93,6 +93,67 @@ def evaluate_scipy(self, points):
         return density_values
 
 
+    def plot_2d_contour(self, dim1, dim2, slice_dims=None, slice_values=None, num_points=100, **kwargs):
+        """
+        Plot a 2D contour of the KDE with optional slicing along other dimensions.
+
+        Parameters:
+            - dim1, dim2: Dimensions for the plot axes.
+            - slice_dims: Dimensions to slice along (list or tuple).
+            - slice_values: Values for slicing along slice_dims (list or tuple).
+            - num_points: Number of points for the contour plot.
+            - **kwargs: Additional keyword arguments passed to the `contour` function.
+        Example:
+            np.random.seed(42)
+            # Number of data points
+            num_points = 1000
+            # Mean and covariance matrix
+            mean = [0, 0, 0]
+            covariance_matrix = [[1, 0.5, 0.3],
+                                [0.5, 1, 0.2],
+                                [0.3, 0.2, 1]]
+
+            # Generate 3D normal distributed data
+            data = np.random.multivariate_normal(mean, covariance_matrix, num_points)
+            parameter = ['m1', 'm2', 'Mc']
+            # Extract individual variables
+            x, y, z = data[:, 0], data[:, 1], data[:, 2]
+            kde = (data)
+
+            # Plot a 2D contour with a slice along the 'z' dimension
+            kde.plot_2d_contour(parameter[0], parameter[1], slice_dims=[parameter[2]], slice_values=[0], num_points=100, colors='blue', alpha=0.5)
+
+        """
+        # Generate a grid for the contour plot
+        dim1_min, dim1_max = np.min(self.data[dim1]), np.max(self.data[dim1])
+        dim2_min, dim2_max = np.min(self.data[dim2]), np.max(self.data[dim2])
+        dim1_grid = np.linspace(dim1_min, dim1_max, num_points)
+        dim2_grid = np.linspace(dim2_min, dim2_max, num_points)
+        xx, yy = np.meshgrid(dim1_grid, dim2_grid)
+        positions = np.column_stack([xx.ravel(), yy.ravel()])
+
+        # If slicing is specified, insert the slice values into the positions array
+        if slice_dims is not None and slice_values is not None:
+            for slice_dim, slice_value in zip(slice_dims, slice_values):
+                slice_idx = list(self.data.columns).index(slice_dim)
+                positions = np.insert(positions, slice_idx, slice_value, axis=0)
+
+        # Evaluate the KDE at the grid points need fix here
```

Comment:
why does the comment say 'need fix here'? 
```suggestion
        # Evaluate the KDE at the grid points
```